### PR TITLE
Finalize data structure

### LINKED ISSUE #6: Finalize parsed project data structure
Review the initial data structure drafted up in https://docs.google.com/document/d/1JP7_OXe4jHicPi7nFR5-Fpmg5fyiGsBBH-0lkSVPyO8/edit
and add the finalized structure to the README

### PR DESCRIPTION
**Pull request by ZhiHanL**
closes #6 

Let me know what you all think of this approach!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,3 +13,35 @@ Follow these steps to set up the project:
 To start the program: `./mvnw spring-boot:run`
 
 If you are using IntellJ IDEA, please install this plugin that provides formatter support: https://github.com/spring-io/spring-javaformat#intellij-idea
+
+## Java Project Custom Data Structure
+Use JavaParser and convert the outputted AST into the custom JavaProject class described below.
+
+### JavaProject Class
+Contains a mapping of package name to class names   
+Contains a mapping of class names to JavaClass classes
```
Comment (algobot76) [inline README.md:22]: I think it would look better if the keywords, such as JavaClass, are highlighted as `JavaClass`.

**Comment by algobot76** [comment]
Also, we should provide an example in JSON format, which will be used by the frontend to generate islands.

**Comment by algobot76** [comment]
This is my suggestion on the JSON example:

```
{
  "packages": [
    {
      "package_name": "p1",
      "classes": [
        {
          "class_name": "c1",
          ...
        }
        ]
    }
    
    ]
}

```

File: README.md [hunk 2]
```diff
@@ -13,3 +13,75 @@ Follow these steps to set up the project:
...
+            "name": "method1",
+            "accessModifier": "private",
+            "parameters": {"param1": "String", "param2": "int"},
+            "returnType": "void"
+          }],
+          "constructors: [{
+            "name": "ex1",
+            "accessModifier": "public",
+            "parameters": {"param1": "String", "param2": "int"},
+            "returnType": "void"
```
Comment (algobot76) [inline README.md:55]: Java constructors don't have any return type.

Comment (ZhiHanL) [reply README.md:55]: true, I guess null would make more sense

**Pull request merged**